apis/core/v1alpha2/core_types.go

```diff
@@ -494,8 +494,50 @@ type DataOutput struct {
 	// If no conditions is specified, it is by default to check output value not empty.
 	// +optional
 	Conditions []ConditionRequirement `json:"conditions,omitempty"`
+	// OutputToReference specifies the object used to store of intermediate data genereted by Operations
+	OutputToReference DataReference `json:"outputToRef,omitempty"`
 }
 
+// DataReference specifies the referenced object in DataOutput or DataInput
+type DataReference struct {
+	runtimev1alpha1.TypedReference `json:",inline"`
+	// Operations specify the data processing operations
+	Operations []DataOperation `json:"operations,omitempty"`
+}
+
+// DataOperation defines the specific operation for data
+type DataOperation struct {
+	// TypeName specific the type of DataOperation
+	TypeName string `json:"type"`
+	// Operator specific the operation under this DataOperation type
+	Operator DataOperator `json:"op"`
+	// FieldPath refers to the value of an object's field.
+	ToFieldPath string `json:"toFieldPath"`
```

Comment:
Just `fieldPath`

Reply:
ToFieldPath is used in DataOperation to distinguish it from the FieldPath in ValueFrom. ToFieldPath means writing data to this filedPath.